main/src/library/MutList.flix

```diff
@@ -702,33 +728,33 @@ namespace MutList {
     ///
     /// The content of the mutable list is unchanged.
     ///
-    pub def reserve!(n: Int32, v: MutList[a]): Unit & Impure =
+    pub def reserve!(n: Int32, v: MutList[a, r]): Unit \ { Read(r), Write(r) } =
         let MutList(a, l) = v;
         a := Array.copyOfRange(0, (deref l) + n, deref a)
 
     ///
     /// Returns `v` as an immutable list.
     ///
-    pub def toList(v: MutList[a]): List[a] & Impure =
+    pub def toList(v: MutList[a, r]): List[a] \ Read(r) =
         foldRight((x, acc) -> x :: acc, Nil, v)
 
     ///
     /// Returns `v` as an array.
     ///
-    pub def toArray(v: MutList[a]): Array[a] & Impure =
+    pub def toArray(v: MutList[a, r]): ScopedArray[a, r] \ { Read(r), Write(r) } =
         let MutList(a, l) = v;
-        Array.copyOfRange(0, deref l, deref a)
+        Array.copyOfRange(0, deref l, deref a) @ Scoped.regionOf(v) // TODO: Remove this region declaration when copyOfRange has been refactored
```

Comment:
This does not work because `@` is only available on specific constructs, not on general function calls. Thus, in this case, you will have to use a cast.